server/config/options.go

```diff
@@ -51,19 +53,25 @@ type CdcConfig struct {
 }
 
 type TracingConfig struct {
-	Enabled           bool
-	WithUDS           string
-	WithAgentAddr     string
-	WithDogStatsdAddr string
+	Enabled           bool   `mapstructure:"enabled" yaml:"enabled" json:"enabled"`
+	WithUDS           string `mapstructure:"agent_socket" yaml:"agent_socket" json:"agent_socket"`
+	WithAgentAddr     string `mapstructure:"agent_addr" yaml:"agent_addr" json:"agent_addr"`
+	WithDogStatsdAddr string `mapstructure:"dogstatsd_addr" yaml:"dogstatsd_addr" json:"dogstatsd_addr"`
 }
 
 type ProfilingConfig struct {
-	Enabled         bool
-	EnableCPU       bool
-	EnableHeap      bool
-	EnableBlock     bool
-	EnableMutex     bool
-	EnableGoroutine bool
+	Enabled         bool `mapstructure:"enabled" yaml:"enabled" json:"enabled"`
+	EnableCPU       bool `mapstructure:"enable_cpu" yaml:"enable_cpu" json:"enable_cpu"`
+	EnableHeap      bool `mapstructure:"enable_heap" yaml:"enable_heap" json:"enable_heap"`
+	EnableBlock     bool `mapstructure:"enable_block" yaml:"enable_block" json:"enable_block"`
```

Comment:
should we remove these keys? and just have Enabled?

Reply:
I would keep them as options for configuration. Some of these are high overhead but could come handy during troubleshooting.

Reply:
Sure, but in most cases, we always want these five to be enabled.

Reply:
For Production deployments I'd only keep CPU and HEAP enabled, these are what I'd call our safe defaults for performance reasons. For development / local deployments, I'd enable all five, due to the overhead they incur. This means we could disable the switches for CPU and HEAP as they will be in most deployments and keep the "extras" configurable if you think that's a better approach.

Reply:
So there are two parts to it I think, 

- Go Pprof: https://pkg.go.dev/net/http/pprof
- Datadog Profiler

I am assuming when you say expensive it is the datadog profiling? If yes, that makes sense. Is that also enabling go pprof by default? or do we need to do something to enable it? For Go pprof, we can enable all and for datadog profiler, we can go with these config knobs.

Reply:
Ah, I believe I now see where you are coming from. Thanks for the extra context! 

Yes, what I said was applicable only for Datadog profiling. 

To confirm if I got your point correctly, let me ask back if I understood you right! Are we suggesting that the tracing and profiling configuration should be generic options and we should figure out what is applicable for the particular deployment? In other words, if we enable CPU profiling, it could enable Datadog CPU Profiling, `go pprof` CPU profiling and other frameworks?

Reply:
So the Go's pprof is cheap as the handler only gets executed when the request is made for the profiling. I would suggest keeping these options only for the datadog related profiling(which is what you have now) and regarding the go proof once we add it will be enabled for profiling.